frontend/public/components/utils/list-dropdown.tsx

```diff
@@ -0,0 +1,293 @@
+    );
+
+    setItems(sortedList);
+
+    // Only update selectedKey if props.selectedKey is defined
+    const newSelectedKey =
+      props.selectedKey !== undefined
+        ? getKey(props.selectedKey, props.selectedKeyKind)
+        : selectedKey;
+
+    setSelectedKey(newSelectedKey);
+
+    const selectedItem = sortedList[newSelectedKey];
+
+    setTitle(
+      selectedItem ? (
+        <ResourceName kind={selectedItem.kindLabel} name={selectedItem.name} />
+      ) : (
+        placeholder
+      ),
+    );
+  }, [
+    loaded,
+    loadError,
+    resources,
+    dataFilter,
+    props.selectedKey,
+    props.selectedKeyKind,
+    placeholder,
+    desc,
+    t,
+    selectedKey,
+  ]);
+
+  const renderedItems = React.useMemo(() => {
+    const result = {};
+    _.keys(items).forEach((key) => {
+      const item = items[key];
+      result[key] = <ResourceName kind={item.kindLabel} name={item.name} />;
+    });
+    return result;
+  }, [items]);
+
+  return (
+    <div>
+      {fixed ? (
+        renderedItems[selectedKey]
+      ) : (
+        <ConsoleSelect
+          actionItems={actionItems}
+          autocompleteFilter={autocompleteFilter}
+          autocompletePlaceholder={placeholder}
+          items={renderedItems}
+          selectedKey={selectedKey}
+          title={loaded ? title : <LoadingInline />}
+          titleIsPlaceholder={false}
+          onChange={handleOnChange}
+          id={id}
+          isFullWidth
+          menuClassName="dropdown-menu--text-wrap"
+          disabled={disabled}
+          dataTest={dataTest}
+        />
+      )}
+      {loaded && _.isEmpty(renderedItems) && (desc || props.selectedKeyKind) && (
+        <Alert
+          isInline
+          className="co-alert pf-v6-c-alert--top-margin"
+          variant="info"
+          title={t('public~No {{selection}} found', {
+            selection: desc || props.selectedKeyKind,
+          })}
+        />
+      )}
+    </div>
+  );
+};
+
+export const ListDropdown: React.FCC<ListDropdownProps> = (props) => {
+  const resources = _.map(props.resources, (resource) =>
+    _.assign({ isList: true, prop: resource.kind }, resource),
+  );
+  return (
+    <Firehose resources={resources}>
+      <ListDropdown_ {...props} />
+    </Firehose>
+  );
+};
+
+export const useProjectOrNamespaceModel = (): [K8sModel, boolean] | [] => {
+  const openshiftFlag = useFlag(FLAGS.OPENSHIFT);
+  const canCreateNs = useFlag(FLAGS.CAN_CREATE_NS);
+  const canCreateProject = useFlag(FLAGS.CAN_CREATE_PROJECT);
+
+  if (flagPending(openshiftFlag) || flagPending(canCreateNs) || flagPending(canCreateProject)) {
+    return [];
+  }
+
+  // NamespaceModal is used when not on an openshift cluster
+  const model = openshiftFlag ? ProjectModel : NamespaceModel;
+  const canCreate = openshiftFlag ? canCreateProject : canCreateNs;
+  return [model, canCreate];
+};
+
+export const NsDropdown: React.FCC<ListDropdownProps> = (props) => {
```

Comment:
Select Namespace has a couple of issues. 

1. The tab navigation focus does not move from "Create Project" into the list of projects. It moves to the browser url bar.
2. The `autocompleteFilter` resets the input text and the list of projects every ~10s

https://github.com/user-attachments/assets/7caa3474-dfd7-4bce-8306-659ee87cabdb

Reply:
1. from "Create project," you need to use the arrow keys to navigate to different items in the menu. This appears to be patternfly as two `DropdownGroup`s work the same way: https://www.patternfly.org/components/menus/custom-menus#with-favorites

2. good catch! this should be fixed now